### DISCUSSION TITLE
Stick client to a specific worker

### DISCUSSION THREAD

**Discussion by dmitry** [puma/puma#3199] [Q&A] [answered]
Is there a way to a stick client via cookies or one specific header in the request to a specific worker?
Use case: I need to create a worker, where each of them is going to be bounded to a specific database (postgres, redis etc), and also VCR is going to be triggered remotely for a single spec. Going to be used for the cypress specs.

**Comment by dentarg** [comment]
Use single mode?

**Reply by dmitry** [reply]
@dentarg do you mean set 1 worker and 1 thread? Seems like that is the only solution due now, but in that case, I have to use different ports to follow the test runners to a different instance. Seems like going to follow this strategy for now.

**Reply by dentarg** [reply]
No, single mode is 0 workers: https://github.com/puma/puma/blob/99ae85bbbe425d1949f9e8693993499aae02fce4/docs/deployment.md?plain=1#L19

**Comment by dentarg** [comment, marked as answer]
Even if you could stick clients to specific workers, I don't think that can help you, as you can not make workers behave differently? Workers are always the same, serving your app. Am I missing something? Just make your app use a different database depending on some info in the request (query param perhaps)?
